### PR TITLE
Fix skaffold init for multimodule maven projects

### PR DESCRIPTION
**Pull request by TadCordle**
Quick idea to fix https://github.com/GoogleContainerTools/skaffold/issues/2844.

Instead of running `jib:_skaffold-init` separately on every module including the root project, we use an aggregator to run the goal once across all modules. This lets us more easily detect the root project, which helps prevent `skaffold init` from setting the `module` field when we don't want it to.

The problem with this approach is that using an aggregator makes getting the `<to><image>` configuration on sub-projects harder; in this case we can only parse it from the configuration block and ignore system properties.

wdyt?

### CODE REVIEW

**Comment by loosebazooka** [comment]
> The problem with this approach is that using an aggregator makes getting the <to><image> configuration on sub-projects harder; in this case we can only parse it from the configuration block and ignore system properties.

Does the person running skaffold init even have a chance to set the image name via property?

**Comment by TadCordle** [comment]
Maybe not system properties, but pom properties?

**Comment by chanseokoh** [comment]
Ah, so it will be sort of broken when the user uses a pom property? I think using pom properties is common, so that's concerning. Hmm...

**Comment by TadCordle** [comment]
Maybe we can parse the properties block the same way.

**Comment by TadCordle** [comment]
Oh wait we can just use `project.getProperties()`, probably.

**Comment by chanseokoh** [comment]
It's still incomplete in that when the person does `<image>example.com/${image.name}</image>`. Sad!

**Comment by TadCordle** [comment]
Ah right. Maybe we can do the replacement ourselves?

**Comment by chanseokoh** [comment]
That sounds like a tough thing to do. Can there be other alternatives? For example, I noticed `<relativePath/>` is empty in the Spring Boot project. Not exactly sure what the Maven API return for this though. From the [doc](https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Example_1), it is suggested that the value could by default `../pom.xml` if not specified, so if `<relativePath/>` returns an empty string, we could figure that the parent POM doesn't exist locally. And I think a Spring Boot project is a somewhat special case to use a parent POM this way, making the project setup unable to have a custom parent.

```xml
<parent>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-starter-parent</artifactId>
	<version>2.1.8.RELEASE</version>
	<relativePath/> <!-- lookup parent from repository -->
</parent>
```

**Comment by TadCordle** [comment]
I think that can work (barring anymore unusual edge cases), using `getProject().getParent().getFile()`. It at least works with the 4 manual tests cases I've been using (spring initalizr project, a parent + 2 modules, a parent + 1 module, and a single pom project).

**Comment by chanseokoh** [comment]
What does it return for each four cases? And I guess `getProject().getParent().getFile()` is basically what `<relativePath>` and there's no other API for that?

**Comment by TadCordle** [comment]
single pom project: `project.getParent()` is null
Initializr: `project.getParent().getFile()` is null
2 modules: each module returns the path to the parent pom
1 module: same thing

File: jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SkaffoldInitMojo.java [hunk 1]
```diff
@@ -37,12 +38,17 @@
   @Override
   public void execute() throws MojoExecutionException {
     checkJibVersion();
+    MavenProject project = getProject();
+    // Ignore parent projects
+    if (project.getModules().size() > 0) {
```
Comment (chanseokoh) [inline jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SkaffoldInitMojo.java]: So this was another bug, right?

Comment (TadCordle) [reply jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SkaffoldInitMojo.java]: I guess you could call it that. It was something I was aware of but didn't think was necessary to change until now.
